### ISSUE TITLE
Implement Feature Selection and Scaling

### ISSUE THREAD

**Issue by Mohid-Abbas** [Mohid-Abbas/cmapss-iot-predictive-pipeline#3]
## Description
Implemented feature selection with Min-Max scaling for sensor data normalization.

## Implementation Details
- Identified and dropped 7 constant sensors (sensor1, sensor5, sensor6, sensor10, sensor16, sensor18, sensor19)
- Selected 17 relevant features (3 settings + 14 sensors)
- Implemented MinMaxScaler with range (0, 1) for normalization
- Supports both fitting new scaler and using existing scaler for consistency
- Preserves engine_id and cycle columns for tracking
- Returns scaled DataFrame and fitted scaler object

## Code Reference
```python
def prepare_features(df: pd.DataFrame, scaler: MinMaxScaler | None = None) -> tuple[pd.DataFrame, MinMaxScaler]:
    features = df[FEATURE_COLUMNS].copy()
    if scaler is None:
        scaler = MinMaxScaler(feature_range=(0, 1))
        scaled_features = scaler.fit_transform(features)
    else:
        scaled_features = scaler.transform(features)
    df_scaled = df[["engine_id", "cycle"]].copy()
    df_scaled[FEATURE_COLUMNS] = scaled_features
    if "rul" in df.columns:
        df_scaled["rul"] = df["rul"].values
    return df_scaled, scaler
```

## Files Changed
- `src/data_prep.py`

## Status
✅ Completed
